geth-patches/README.md

```diff
@@ -9,44 +9,36 @@ We try to minimize number and amount of changes in those patches as much as poss
- - `0003-dockerfiles-wnode-swarm.patch` — adds Dockerfiles (who uses this?)
- - `0004-whisper-notifications.patch` — adds Whisper notifications (need to be reviewed and documented)
- - `0006-latest-cht.patch` – updates CHT root hashes, should be updated regularly to keep sync fast, until proper Trusted Checkpoint sync is not implemented as part of LES/2 protocol.
- - `0007-README.patch` — update upstream README.md.
- - `0008-tx-pool-nonce.patch` - On GetTransactionCount request with PendingBlockNumber get the nonce from transaction pool
+ - [`0000-accounts-hd-keys.patch`](./0000-accounts-hd-keys.patch) — adds support for HD extended keys (links/docs?)
+ - [`0002-les-api-status.patch`](./0002-les-api-status.patch) — adds StatusBackend into LES code (need to be inspected, some things can and should be done outside of les code
+ - [`0003-dockerfiles-wnode-swarm.patch`](./0003-dockerfiles-wnode-swarm.patch) — adds Dockerfiles (who uses this?)
+ - [`0004-whisper-notifications.patch`](./0004-whisper-notifications.patch) — adds Whisper notifications (need to be reviewed and documented)
+ - [`0006-latest-cht.patch`](./0006-latest-cht.patch) – updates CHT root hashes, should be updated regularly to keep sync fast, until proper Trusted Checkpoint sync is not implemented as part of LES/2 protocol.
+ - [`0007-README.patch`](./0007-README.patch) — update upstream README.md.
+ - [`0008-tx-pool-nonce.patch`](./0008-tx-pool-nonce.patch) - On GetTransactionCount request with PendingBlockNumber get the nonce from transaction pool
+ - [`0010-geth-17-fix-npe-in-filter-system.patch`](./0010-geth-17-fix-npe-in-filter-system.patch) - Temp patch for 1.7.x to fix a NPE in the filter system
 
 # Updating upstream version
 
 When a new stable release of `go-ethereum` comes out, we need to upgrade our fork and vendored copy.
 
 **Note: The process is completely repeatable, so it's safe to remove current `go-ethereum` directory, clone latest upstream version and apply patches from scratch.**
 
-## How to update forked version
-Make sure you have `status-go` in your `$GOPATH/src/github.com/status-im/` first.
+### Using existing fork repo (recommended)
 
-### From very scratch
-Use this method if you're up to nuke forked repo for some reason and apply patches from scratch:
+#### I. In `go-ethereum fork`.
 
+1. Remove the local `develop` branch.
 ```
-# from scratch
-rm -rf $GOPATH/src/github.com/status-im/go-ethereum
-cd $GOPATH/src/github.com/status-im/
-git clone https://github.com/ethereum/go-ethereum
-
-# update remote url to point to our fork repo
-git remote set-url origin git@github.com:status-im/go-ethereum.git
+git branch -D develop
 ```
 
-### Using existing fork repo (recommended)
-
+2. Pull upstream release branch into `develop` branch.
 ```
-# merge upstream release branch into local master
-git pull git@github.com:ethereum/go-ethereum.git release/1.7:master
+git pull git@github.com:ethereum/go-ethereum.git release/1.7:develop
```

Comment:
We should put a comment saying that upstream release branch depends on the current stable `geth` version we use. Currently, it's `1.7.x`.